### DISCUSSION TITLE
Robust optimisation - adaptive radiotherapy

### DISCUSSION THREAD

**Discussion by bgeorge0** [e0404/matRad#629] [Ideas]
Hi,

I'd like to see about using matRad to test some ideas about robust optimisation/planning for adaptive radiotherapy. I hope I can explain here. 

In the current application of robust optimisation, a single plan is produced which is robust against e.g., patient setup errors, proton range uncertainty, or variations in location of OARs. In the end you produce one set of fluences and MLC positions which make up this plan and are delivered each day.

In adaptive radiotherapy, each fraction may have different variations of patient anatomy, e.g., OAR position, but the adaptive process allow re-optimisation to generate a new set of fluences and MLC positions to account for this. Therefore, in adaptive radiotherapy the component which needs to be 'robust' is not the fluences and MLC positions, but the objectives and constraints used to guide the optimiser.

These two concepts are similar but inherently different. Perhaps the first is 'robust optimisation' and the second is better to be known as 'robust planning'.

I've had a look through the robust optimisation examples, but not sure if the architecture of matRad easily allows this idea of robust optimisation to be extended to 'robust planning'.

Please let me know if this doesn't make any sense or if I've missed something obvious.

**Comment by wahln** [comment]
Hi,

yes, our robust optimization code is purely intended for robustness against uncertainties in range, setup, and anatomical variations, but could also be extended towards other sources of uncertainty. 

Do you have an particular concept/literature reference in mind for adaptation with what you call "robust planning"? There is no inherent support for that. Also I am probably not the biggest expert on adaptive planning, but I think that your idea of "robust planning" is more a mix of choice of objective function (e.g. through knowledge-based planning approaches), optimization method (for example you can use a prioritized goal formalism instead of a weighted sum approach), and so on. Most of these concepts are realizable within matRad and also worked upon in some way or the other.

**Comment by bgeorge0** [comment]
Hi,

Thanks for your quick reply!

I don't have any literature references for this 'robust planning' technique - it is something which I've been thinking about recently but haven't seen discussed or presented elsewhere, so was looking for a platform to do some sort of proof of concept.

I think I need to look at the terminology you use in matRad and make sure I'm using it correctly, but I guess the idea is that you have one set of inputs to your objective function which produce optimal fluences for multiple anatomical variations. Whereas in traditional treatment planning techniques, you would continue to adjust the inputs to your objective function to find the optimal fluences for a specific anatomy.

In _robust planning_ you'd therefore have multiple stf structs (to represent each anatomical variation) each with its own dij matrix, rather than the current _robust optimisation_ approach where you have multiple dij matricies to represent the interaction of a single stf with multiple geometries. You would still do some sort of minmax optimisation on the set dij matries, but you'd return a set of stf structs.

Or maybe I'm not making any sense!

**Comment by wahln** [comment]
As you said, robust optimization will use your given geometry (patient + stf), but then internally create error scenarios for those (e.g. shift in isocenter, another CT with deformations, change in the relative electron densities or relative stopping powers, etc.) to compute a dij for each scenario (the dij for scenarios are stored in the cell array in dij.physicalDose{:}, for example). 
During optimization, in each iteration, all scenario doses will be computed from the fluence vector. Then the chosen strategy will do something with those error scenarios - for example, the COWC strategy will evaluate the composite weighted objective function for each scenario, and minimize the maximum of these values. You will not end up with multiple fluence vectors, but with one fluence vector that is supposed to be _robust_ against the given scenarios. 

So under the hood, you actually have, in a sense, multiple stfs (e.g. shifted isocenter) and/or cts/csts (deformed anatomy), and for each of them a dijs, in robust optimization The key is that all of these represent the same beam arrangement, number of spots, etc. 

So I am not entirely sure, what your multiple stfs in robust planning would do - you mean multiple beam arrangements etc? This would not integrate into a robust optimization approach, because with different beam arrangements etc. you will have fluence vectors of incompatible dimensions. It sounds to me that you would rather have different planning scenarios, and determine the optimal planning scenario for your daily adaptation?

**Comment by bgeorge0** [comment]
The bit which I'd like to change for adaptive planning is based on this: "During optimization, in each iteration, all scenario doses will be computed from the fluence vector". I'd like to have each scenario dose have a different fluence vector associated with it. The situation I'm trying to model is that you acquire a daily image of the patient to determine the anatomy at that point in time, but then get the opportunity to re-optimise your fluence based on that anatomy. Whether you arrive at a clinically acceptable solution is based on how you set up your objectives and constraints. It may be that these need to changed for some particular daily anatomy, e.g., to control unwanted dose splash. The idea I had is that you could use some sort of robust planning to ensure that the chosen objectives and constraints produce clinically acceptable for a variety of anatomical variations.

Plan elements (number of beams, gantry angles, isocentre, etc.) will remain constant throughout, but not sure if that means that fluence vectors will be the same size for each scenario.
